fvm/context.go

```diff
@@ -258,14 +260,24 @@ func WithServiceAccount(enabled bool) Option {
 }
 
 // WithRestrictedDeployment enables or disables restricted contract deployment for a
-// virtual machine context.
+// virtual machine context. Warning! this would be overridden with the flag stored on chain.
+// this is just a fallback value
 func WithRestrictedDeployment(enabled bool) Option {
 	return func(ctx Context) Context {
 		ctx.RestrictedDeploymentEnabled = enabled
 		return ctx
 	}
 }
 
+// WithRestrictedContractRemovalEnabled enables or disables restricted contract removal for a
+// virtual machine context.
+func WithRestrictedContractRemovalEnabled(enabled bool) Option {
```

Comment:
default value is true (restricted) for all networks right now, so no call to this yet.